src/openvpn/buffer.c

```diff
@@ -442,6 +442,7 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
 				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
 				    gc);
   int i;
+  ASSERT( maxoutput > 0 || separator != NULL );
```

Comment:
Hi,

On Tue, Jun 07, 2016 at 01:57:04PM -0700, Ilya Shipitsin wrote:
> > @@ -442,6 +442,7 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
> >  				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
> >  				    gc);
> >    int i;
> > +  ASSERT( maxoutput > 0 || separator != NULL );
> 
> "int i;" always been there

Not that one, the "buffer out = alloc_buf_gc()" bit needs to come after
the ASSERT(), and that means it will have to be

int i;
struct buffer out;
ASSERT(...)
out = alloc_buf_gc(...)

gert
-- 
USENET is *not* the non-clickable part of WWW!
                                                           //www.muc.de/~gert/
Gert Doering - Munich, Germany                             gert@greenie.muc.de
fax: +49-89-35655025                        gert@net.informatik.tu-muenchen.de